src/app/modules/project/details/details.component.html

```diff
@@ -104,4 +104,13 @@ <h1 class="project-title">{{ project.name }}</h1>
       </div>
     </div>
   </div>
-</div>
\ No newline at end of file
+</div>
+
+<ng-template #noProject>
+  <div class="row">
+    <div class="offset-2 col-8">
+      <h1>Project details</h1>
+      <p>No project could be found with id: {{invalidId}}</p>

```

Comment:
Should this not just show the existing not found (404) page?

Reply:
I prefer using the 404 when the url is completly malformed and no relation to any component can be found. In this scenario a component exists for the route however the entered id is malformed. By not redirecting the user to the 404 page I can show them a more specific error instead of 'the requested page could not be found'

Reply:
Okay. Fair point.
I would suggest the following message though (Since id can be quite vague for general users).
```suggestion
      <p>The project with the identifier "{{invalidId}}" could not be found.</p>

```